backupccl: pause online restore if disk is full

dt opened the pull request:
Release note: none.
Epic: none.

dt committed:
kvserver: move MinCapacityForBulkIngest setting to pkg/storage

Release note: none.
Epic: none.

dt committed:
server: fail downloads if disk is too full

Previously download RPCs would continue to request pebble download external
files even to the point where the disk filled to 100% and the node crashed.

Now the download RPC will fail if the disk is approaching full, as defined by
the kv.bulk_io_write.min_capacity_remaining_fraction setting.

Release note: none.
Epic: none.

dt committed:
backupccl: pause rather than fail download job if disk is full

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/124700)
<!-- Reviewable:end -->

msbutler reviewed (commented):
not sure why this doesn't work :D

msbutler commented on pkg/server/span_download.go:
```diff
@@ -87,16 +89,44 @@ func (s *systemStatusServer) localDownloadSpan(
 				return sendDownloadSpans(ctx, req.Spans, downloadSpansCh)
 			},
+			func(ctx context.Context) error {
+				tick := time.NewTicker(time.Second * 15)
```
yeah fine, leave it as is. more trouble than it's worth

dt commented:
> not sure why this doesn't work

Turns out Download communicates all its errors back to its caller in plain untyped strings. Fix incoming.

dt committed:
server: return encoded errors from DownloadSpan

Previously DownloadSpan returned its errors in plain strings in its response,
or as gRPC errors that then were moved into strings in the response by the
errorFn in iterateNodes. This, however, meant these errors lost all of their
type information.

Returning an error to a gRPC method also does not cause it to be encoded and
transported to the caller. Instead, we must encode the error and put the encoded
error in a field of the response, as is done for BatchRequest/Response and in
execinfrapb's FullError field, in the body of the gRPC method's implementation.

This is done here for errors returned by the localDownloadSpan section of the
DownloadSpan RPC implementation, however there are still other error returns
in this RPC before that call that still use the gRPC error return, as to numerous
other RPC methods on the status server and similar RPC APIs, so a broader solution
may be desired here in the future.

Release note: none.
Epic: none.

dt commented:
TFTR!

bors r+
